src/main/java/org/opentripplanner/routing/api/request/RoutingRequest.java

```diff
@@ -805,6 +790,11 @@ public long getSecondsSinceEpoch() {
         return dateTime;
     }
 
+    public ZonedDateTime getZonedDateTime() {
+        var zoneId = rctx.graph.getTimeZone().toZoneId();
```

Comment:
The `rctx` can be null, and passing it around inside the request is really dirty, so I would like to avoid using it - and move in the direction of removing this from the request. I also have a feeling that this duplicate the logic in `DateMapper.asStartOfService`, witch is slightly different. Hence; if people start to use this method that would lead to errors. So, instead of calculating this here, use the `RoutingWorker:searchStartTime`.